src/app/shared/object-collection/shared/badges/access-status-badge/access-status-badge.component.ts

```diff
@@ -96,7 +118,36 @@ export class AccessStatusBadgeComponent implements OnDestroy, OnInit {
     );
   }
 
-  ngOnDestroy(): void {
-    this.subs.filter((sub) => hasValue(sub)).forEach((sub) => sub.unsubscribe());
+  /**
+   * Method to handle the object type Bitstream
+   */
+  private handleBitstream() {
+    this.showAccessStatus = environment.item.bitstream.showAccessStatuses;
+    if (!this.showAccessStatus) {
+      // Do not show the badge if the feature is inactive.
+      return;
+    }
+    const bitstream = this.object as Bitstream;
+    if (bitstream.accessStatus == null) {
+      return;
```

Comment:
This means the badge won't appear if it's ever used on a page that doesn't embed the link...

You can handle this in one go for both Items and Bitstreams:
1. Type the input as `Item | Bitstream` (this will also make it more clear which DSOs it can be used with)
2. Check `hasValue(object.accessStatus)` in `ngOnInit`
3. If it's empty, call `LinkService.resolveLink(this.object, followLink('accessStatus'))`

This would be a little slower than with embeds, but better than losing the badge completely

Reply:
OK, I didn't knew for discriminated unions. I also changed the code to use the resolveLink as well for items.